src/sugar3/graphics/popwindow.py

```diff
@@ -0,0 +1,204 @@
+
+        Returns:
+            None
+        '''
+        self.destroy()
+
+    def __key_press_event_cb(self, window, event):
+        '''
+        key-press-event signal handler
+        Destroys the pop-up window on pressing Escape.
+
+        Args:
+           window (Gtk.Widget): the window which received the signal
+           event (Gdk.EventKey): the event which triggered the signal
+
+        Returns:
+            None
+        '''
+        keyname = Gdk.keyval_name(event.keyval)
+        if keyname == 'Escape':
+            self.destroy()
+
+    def __realize_cb(self, widget):
+        '''
+        realize signal handler
+        Sets window as a transient window for the window specified by parent.
+
+        Args:
+            widget (Gtk.Widget): the widget(window) which received the signal
+
+        Returns:
+            None
+        '''
+        self.set_type_hint(Gdk.WindowTypeHint.DIALOG)
+        window = self.get_window()
+        window.set_accept_focus(True)
+
+        if self._parent_window_xid is not None:
+            display = Gdk.Display.get_default()
+            parent = GdkX11.X11Window.foreign_new_for_display(
+                display, self._parent_window_xid)
+            window.set_transient_for(parent)
+            shell.get_model().push_modal()
+
+    def __hide_cb(self, widget):
+        '''
+        hide signal handler
+
+        Args:
+            widget (Gtk.Widget): the widget(window) which received the signal
+
+        Returns:
+            None
+        '''
+        shell.get_model().pop_modal()
+
+    def add_view(self, widget, expand=True, fill=True, padding=0):
+        '''
+        Adds child to the vbox.
+
+        Args:
+            widget (Gtk.Widget): widget to be added
+
+            expand (bool): True if child is to be given extra space allocated
+                to vbox.
+
+            fill (bool): True if space given to child by the expand option is
+                actually allocated to child, rather than just padding it.
+
+            padding (int): extra space in pixels to put between child and its
+                neighbors, over and above the global amount specified
+                by spacing in vbox.
+
+        Returns:
+            None
+        '''
+        self._vbox.pack_start(widget, True, fill, padding)
+
+
+class TitleBox(ToolbarBox):
+    '''
+    Title box at the top of the pop-up window.
+    Title and close button are added to the box and as needed more widgets
+    can be added using self.add_widget method.
+    This box is optional as the inherited class can remove this block by
+    setting the self._set_title_box to False.
+    '''
+    def __init__(self, title):
+        ToolbarBox.__init__(self)
+        self.close_button = ToolButton(icon_name='dialog-cancel')
+        self.toolbar.insert(self.close_button, -1)
+        self.close_button.show()
+
+        label = Gtk.Label()
+        label.set_markup('<b>%s</b>' % title)
```

Comment:
Yes done :)
Have a look at the example of this https://github.com/sugarlabs/sugar/pull/672/files#diff-52320dfca28b62095ceb6c9c63c87b9dR322